pennylane/ops/op_math/decompositions/controlled_decompositions.py

```diff
@@ -523,30 +546,48 @@ def _mcx_two_workers(wires, work_wires, work_wire_type, **__):
         `arXiv:2407.17966 <https://arxiv.org/abs/2407.17966>`__
 
     """
-
+    # Unpack work wires for readability. There might just be one of them if it is a "small" MCX
+    # (less than 6 controls)
+    work0, *work1 = work_wires
     # First use the work wire to prepare the first two control wires as conditionally clean.
-    ops.Toffoli([wires[0], wires[1], work_wires[0]])
+    left_elbow = ops.Toffoli if work_wire_type == "borrowed" else qml.TemporaryAND
+    left_elbow([wires[0], wires[1], work0])
+
     middle_ctrl_indices = _build_log_n_depth_ccx_ladder(wires[:-1])
 
-    # Apply the MCX in the middle
+    # Apply the MCX in the middle. This is just a single Toffoli without work wires for "small" MCX
     if len(middle_ctrl_indices) == 1:
-        ops.Toffoli([work_wires[0], wires[middle_ctrl_indices[0]], wires[-1]])
+        ops.Toffoli([work0, wires[middle_ctrl_indices[0]], wires[-1]])
     else:
         middle_wires = [wires[i] for i in middle_ctrl_indices]
-        _mcx_one_worker(work_wires[:1] + middle_wires + wires[-1:], work_wires[1:])
+        # No toggle detection needed for the inner MCX decomposition, even for borrowed work wires
+        _mcx_one_worker(
+            [work0] + middle_wires + wires[-1:],
+            work1,
+            work_wire_type=work_wire_type,
+            _skip_toggle_detection=True,
```

Comment:
Passing the `work_wire_type` onwards here allows `_mcx_one_worker` to use `TemporaryAND` itself if the type is `"zeroed"`. Previously, we were always using `"zeroed"` (the default) by not passing `work_wire_type` on, because this was only deactivating toggle detection. Instead, we now have an explicit kwarg to switch off toggle detection.